examples/llama3-vllm-gcp/llama3_vllm_gcp.py

```diff
@@ -0,0 +1,192 @@
+        # (Optional) Include the following to create exposed TLS endpoints:
+        # open_ports=[443], # Expose HTTPS port to public
+        # server_connection_type="tls", # Specify how runhouse communicates with this cluster
+        # den_auth=False, # No authentication required to hit this cluster (NOT recommended)
+    )
+
+    # We'll set an `autostop_mins` of 30 for this example. If you'd like your cluster to run indefinitely, set `autostop_mins=-1`.
+    # You can use SkyPilot in the terminal to manage your active clusters with `sky status` and `sky down <cluster_id>`.
+    #
+    # Next, we define the environment for our module. This includes the required dependencies that need
+    # to be installed on the remote machine, as well as any secrets that need to be synced up from local to remote.
+    # Passing `huggingface` to the `secrets` parameter will load the Hugging Face token we set up earlier.
+    #
+    # Learn more in the [Runhouse docs on envs](/docs/tutorials/api-envs).
+    env = rh.env(
+        reqs=["vllm==0.2.7"],  # >=0.3.0 causes Pydantic version error
+        secrets=["huggingface"],  # Needed to download Llama 3 from HuggingFace
+        name="llama3inference",
+        working_dir="./",
+    )
+
+    # Finally, we define our module and run it on the remote cluster. We construct it normally and then call
+    # `get_or_to` to run it on the remote cluster. Using `get_or_to` allows us to load the exiting Module
+    # by the name `llama3-8b-model` if it was already put on the cluster. If we want to update the module each
+    # time we run this script, we can use `to` instead of `get_or_to`.
+    #
+    # Note that we also pass the `env` object to the `get_or_to` method, which will ensure that the environment is
+    # set up on the remote machine before the module is run.
+    remote_llama_model = LlamaModel().get_or_to(
+        gpu_cluster, env=env, name="llama3-8b-model"
+    )
+
+    # ## Calling our remote function
+    #
+    # We can call the `generate` method on the model class instance as if it were running locally.
+    # This will run the function on the remote cluster and return the response to our local machine automatically.
+    # Further calls will also run on the remote machine, and maintain state that was updated between calls, like
+    # `self.engine`.
+    prompt = "The best chocolate chip cookie is"
+    ans = await remote_llama_model.generate(
+        prompt=prompt, temperature=0.8, top_p=0.95, max_tokens=100
+    )
+    for text_output in ans:
+        print(f"... Generated Text:\n{prompt}{text_output}\n")
+
+    # :::note{.info title="Note"}
+    # Your initial run of this script may take a few minutes to deploy an instance on GCP, set up the environment,
+    # and load the Llama 3 model. Subsequent runs will reuse the cluster and generally take seconds.
+    # :::
+
+    # ### Advanced: Sharing and TLS endpoints
```

Comment:
do we want to include this? It seems overkill / could include in a different example, but I'm fine with it if you think so

Reply:
It seemed like something missing from the other examples to me and possibly a nature way to highlight the value of Den to new users - I think it's easy to miss out on that value entirely. We could experiment with it here and remove later if we have more/better ways to introduce sharing/den in the examples?